app/routes/start-application.js

```diff
@@ -23,22 +29,38 @@ module.exports = [{
   path: '/start-application',
   options: {
     validate: {
-      payload: schema
+      payload: Joi.object()
+        .concat(sbiSchema)
+        .concat(agreementNumberSchema)
     },
     handler: async (request, h) => {
-      let { agreement, data, callerId, crn } = await cache.get(request)
+      let agreement
+      const { data, callerId, crn } = await cache.get(request)
+      const sbi = parseInt(request.payload.sbi)
 
-      // if no existing agreement or SBI no longer matches cached agreement then start new agreement
-      if (!agreement || agreement?.organisation?.sbi !== request.payload.sbi) {
-        await cache.reset(request)
+      await cache.reset(request)
+      if (!request.payload.agreementNumber) {
         agreement = create()
-        // Need to include caller Id as part of agreement to support downstream services
-        // Expect to remove once Defra Identity integrated
-        agreement.callerId = callerId
-        agreement.crn = crn
-      }
+      } else {
+        const incompleteApplications = await getBySbi(sbi)
+        const agreements = incompleteApplications.agreements
```

Comment:
Consider `const { agreements } = incompleteApplications` for consistency

(I also thought doing this eliminated silent issues as `const { notInTheObject } = anObject` would error saying does not exist but does not , returns undefined which then when you try to expand errors, sad times 😢 )

Reply:
changed